src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -971,10 +972,11 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
         //          > val vparamSymssOrEmptyParamsFromOverride =
         //          This means an accessor that overrides a Java-defined method gets a MethodType instead of a NullaryMethodType, which breaks lots of assumptions about accessors)
         def checkCanAutoApply(): Boolean = {
-          if (sourceLevel2_14 && !meth.isJavaDefined)
+          if (sourceLevel2_14 && !meth.isJavaDefined && !isAnyDerived(meth)) {
             context.deprecationWarning(tree.pos, NoSymbol, s"Auto-application to `()` is deprecated. Supply the empty argument list `()` explicitly to invoke method ${meth.decodedName},\n" +
                                                            s"or remove the empty argument list from its definition (Java-defined methods are exempt).\n"+
```

Comment:
You could fix the wording here to "immediate overrides that are Java-defined and members of Any (if you're seeing this message then something is broken)."